src/storage/src/store_impl.rs

```diff
@@ -75,6 +78,27 @@ fn may_dynamic_dispatch(
     }
 }
 
+fn may_verify(state_store: impl StateStore + AsHummockTrait) -> impl StateStore + AsHummockTrait {
+    #[cfg(not(debug_assertions))]
+    {
+        state_store
+    }
+    #[cfg(debug_assertions)]
+    {
+        let expected = if env_var_is_true("ENABLE_STATE_STORE_VERIFY") {
+            info!("enable verify state store");
+            Some(SledStateStore::new_temp())
+        } else {
+            info!("verify state store is not enabled");
+            None
```

Comment:
Is it better to just return `state_store` directly instead of instantiating `VerifyStateStore` with `expected: None` if `ENABLE_STATE_STORE_VERIFY` is not set?

Reply:
We can also doing None check for expected in the VerifyStateStore implementation.

Reply:
We cannot return state store directly, since the env var check is done at runtime, while the type of return value is determined statically at compile time. Therefore, under debug mode, the return type is always `VerifyStateStore`, and we can only use `Option` to indicate whether the verification is enabled after it is instantiated.

Reply:
Ah I see. I ignore the fact that we are not using Box dyn here. Thanks for the explanation.